### PR TITLE
updated for MATLAB 2017a

### PR DESCRIPTION
**Pull request by hajgato**
Matlab 2017a install complains that you should not run the installed from the root directory of install CD.

### CODE REVIEW

File: easybuild/easyblocks/m/matlab.py [hunk 1]
```diff
@@ -107,14 +109,27 @@ def install_step(self):
         # make sure install script is executable
         adjust_permissions(src, stat.S_IXUSR)
 
+        if LooseVersion(self.version) >= LooseVersion('2017a'):
+            jdir = os.path.join(self.cfg['start_dir'], 'sys', 'java', 'jre', 'glnxa64', 'jre', 'bin')
+            for perm_file in [os.path.join(self.cfg['start_dir'], 'bin', 'glnxa64'), jdir]:
```
Comment (wpoely86) [inline easybuild/easyblocks/m/matlab.py]: don't you mean `perm_dir` here? It's a whole directory?

File: easybuild/easyblocks/m/matlab.py [hunk 2]
```diff
@@ -107,14 +109,27 @@ def install_step(self):
...
         if not '_JAVA_OPTIONS' in self.cfg['preinstallopts']:
             self.cfg['preinstallopts'] = ('export _JAVA_OPTIONS="%s" && ' % self.cfg['java_options']) + self.cfg['preinstallopts']
-        cmd = "%s ./install -v -inputFile %s %s" % (self.cfg['preinstallopts'], self.configfile, self.cfg['installopts'])
+        chdircmd = ' '
+        installerpath = '.'
+        if LooseVersion(self.version) >= LooseVersion('2017a'):
+            chdircmd = 'cd %s &&' % self.builddir 
+            installerpath = self.cfg['start_dir']
+
+        cmd = "%s %s %s/install -v -inputFile %s %s" % (self.cfg['preinstallopts'], chdircmd, installerpath, self.configfile,
```
Comment (wpoely86) [inline easybuild/easyblocks/m/matlab.py]: I wouldn't do it like this. Do the chdir before getting to this line (in the if). Mixing in the `chdircmd` and `preinstallopts` might do weird things.

Comment (hajgato) [inline easybuild/easyblocks/m/matlab.py]: @wpoely86 fixed

File: easybuild/easyblocks/m/matlab.py [hunk 3]
```diff
@@ -107,14 +109,22 @@ def install_step(self):
...
         # make sure $DISPLAY is not defined, which may lead to (hard to trace) problems
         # this is a workaround for not being able to specify --nodisplay to the install scripts
         if 'DISPLAY' in os.environ:
             os.environ.pop('DISPLAY')
 
         if not '_JAVA_OPTIONS' in self.cfg['preinstallopts']:
             self.cfg['preinstallopts'] = ('export _JAVA_OPTIONS="%s" && ' % self.cfg['java_options']) + self.cfg['preinstallopts']
-        cmd = "%s ./install -v -inputFile %s %s" % (self.cfg['preinstallopts'], self.configfile, self.cfg['installopts'])
+        if LooseVersion(self.version) >= LooseVersion('2017a'):
+            run_cmd('cd %s' % self.builddir, log_all=True, simple=True)
```
Comment (wpoely86) [inline easybuild/easyblocks/m/matlab.py]: use `change_dir` from `easybuild.tools.filetools`

Comment (hajgato) [inline easybuild/easyblocks/m/matlab.py]: @wpoely86 Done

**Comment by boegel** [comment]
lgtm, tested with https://github.com/hpcugent/easybuild-easyconfigs/pull/4616, good to go, thanks @hajgato!

**Pull request merged**